includes/class-sensei-admin.php

```diff
@@ -525,6 +525,17 @@ public function duplicate_course_with_lessons_action() {
 		$this->duplicate_content( 'course', true );
 	}
 
+	/**
+	 * Redirect the user safely
+	 *
+	 * @param  string $redirect_url URL to redirect the user.
+	 * @return void
+	 */
+	public function safe_redirect( $redirect_url ) {
```

Comment:
As a general rule, I'm not a fan of making a method like this public. I know that we're doing so because we need to mock it in the test, but I think this is an anti-pattern and should be avoided if possible.

If it cannot be avoided, we should at least add `@access private` to the doc comment above to mark this as a method that is not part of Sensei's public API.

Reply:
+1 to not making this `public` if we can help it.

Reply:
I agree with y'all, but I searched for some ways to mock this function, and I didn't find anything. I inspected the following functions:

https://developer.wordpress.org/reference/functions/wp_safe_redirect/#source
https://developer.wordpress.org/reference/functions/wp_redirect/#source

But nothing seems to be possible to mock. Some idea? The error that I get if I don't mock it is: `Cannot modify header information - headers already sent by...`

Meanwhile, I'm adding the `@access private`

Reply:
At a minimum, it could be changed to `protected`. I don't think there's any reason to make it fully `public`.

Reply:
Makes totally sense! I didn't know that we could mock protected methods too =)
Doing it!